apps/src/code-studio/components/aiTutor/compilationTutor.tsx

```diff
@@ -1,39 +1,49 @@
 import React from 'react';
-import PropTypes from 'prop-types';
-import {useSelector, useDispatch} from 'react-redux';
+import {useSelector} from 'react-redux';
 import Button from '@cdo/apps/templates/Button';
 import style from './ai-tutor.module.scss';
-import {askAITutor} from '@cdo/apps/aiTutor/redux/aiTutorRedux';
+import {AITutorState, askAITutor} from '@cdo/apps/aiTutor/redux/aiTutorRedux';
+import {JavalabEditorState} from '@cdo/apps/javalab/redux/editorRedux';
+import {JavalabState} from '@cdo/apps/javalab/redux/javalabRedux';
 import analyticsReporter from '@cdo/apps/lib/util/AnalyticsReporter';
 import {EVENTS} from '@cdo/apps/lib/util/AnalyticsConstants';
 import {compilationSystemPrompt} from '@cdo/apps/aiTutor/constants';
+import {useAppDispatch} from '@cdo/apps/util/reduxHooks';
 
 // AI Tutor feature that explains to students why their code did not compile.
-const CompilationTutor = ({levelId}) => {
-  const dispatch = useDispatch();
+const CompilationTutor = (levelId: number) => {
+  const dispatch = useAppDispatch();
 
-  const sources = useSelector(state => state.javalabEditor.sources);
-  const fileMetadata = useSelector(state => state.javalabEditor.fileMetadata);
-  const activeTabKey = useSelector(state => state.javalabEditor.activeTabKey);
+  const sources = useSelector(
+    (state: {javalabEditor: JavalabEditorState}) => state.javalabEditor.sources
```

Comment:
Something I've been using recently is `TypedUseSelectorHook`, which allows you to create a `useSelector` with the typings on the state object that you need, for example
```
const useTypedSelector: TypedUseSelectorHook<{javalabEditor: JavalabEditorState; aiTutor: AITutorState}> = useSelector;
...

// knows what type state is automatically
const sources = useTypedSelector(state => state.javalabEditor.sources);
const aiResponse = useTypedSelector(state => state.aiTutor.aiResponse);
```
It can be overkill sometimes but I find it helpful when you have a lot of `useSelectors`. Not a requirement, but just thought i'd share if you find it useful!

(Also thinking out loud a bit, if folks do find it useful, we could even vend out typed useSelector hooks with each redux store, like `useJavalabEditorSelector`, `useAiTutorSelector`, etc, that are automatically typed...though I suppose this would only be useful until we have a global store type. @molly-moen curious what you think though)

Reply:
So after reviewing this PR I thought to myself "why don't we just make a global type for our redux store" and started doing that 😆 . I should have a PR pretty shortly with the creation of that type, after seeing this I'll also add a `TypedUseSelectorHook` that we can use too!

Reply:
https://github.com/code-dot-org/code-dot-org/pull/56208

Reply:
(not blocking on this pr, we can follow up and clean this up if desired)

Reply:
Ooooh. I like the look of this. Seems much more streamlined. I'm going to skip incorporating on this PR because I'm in a PR domino situation and want to unblock myself from merge conflicts on other PRs 🙃, but made a note to update on my "Future AI Tutor work items" list. Thank you, both!